### PR TITLE
Add initial adaptive refinement

### PR DESCRIPTION
**Pull request by shahabgol**
# Description of the problem

- Choosing a large number of mesh initial refinement makes the code very slow in the first step (before adaptive refinement can make the grids larger in unnecessary regions).

# Description of the solution

- Adds initial (pre-solve) adaptive mesh refinement.

### CODE REVIEW

**Review by tonielgeitani — APPROVED** [review]

File: source/core/parameters.cc [hunk 1]
```diff
@@ -1531,6 +1531,11 @@ namespace Parameters
   {
     prm.enter_subsection("mesh adaptation");
     {
+      prm.declare_entry("initial refinement steps",
```
Comment (Luckabarbeau) [inline source/core/parameters.cc:1534]: Number of initial refinement step would be better I think

**Review by Luckabarbeau — COMMENTED** [review]
very nice and short Pr i would also implement this in all the other solver

File: source/solvers/gd_navier_stokes.cc [hunk 2]
```diff
@@ -1215,7 +1215,17 @@ GDNavierStokesSolver<dim>::solve()
...
+        {
+          for (unsigned int i = 0;
+               i <
+               this->simulation_parameters.mesh_adaptation.initial_refinement;
+               i++)
+            NavierStokesBase<dim,
+                             TrilinosWrappers::MPI::BlockVector,
+                             std::vector<IndexSet>>::refine_mesh();
+
+          this->iterate();
```
Comment (blaisb) [inline source/solvers/gd_navier_stokes.cc:1227]: This is a good idea really, but I suggest you move it to the initial condition. This way, for example, you could re-apply the initial condition between every mesh refinement. In addition, this way every solver would have this capability automatically.
I really suggest moving it to initial conditions in the NavierStokesBase and to reapply the initial condition every time.

**Review by blaisb — CHANGES_REQUESTED** [review]
One major suggestion + don't forget to add documentation to .rst file :)

File: source/core/parameters.cc (hunk 1 above)
Comment (shahabgol) [reply source/core/parameters.cc:1534]: I think a little bit long for a parameter name

**Review by blaisb — APPROVED** [review]

**Pull request merged**